src/js/core/widget/core/MainTab.js

```diff
@@ -0,0 +1,264 @@
+/**
+ * #Main Tab
+ * Main Tab component is dedicated to control page change in app.
+ * The component is positioned always on bottom of the screen.
+ *
+ * @class ns.widget.core.MainTab
+ * @extends ns.widget.BaseWidget
+ * @author Tomasz Lukawski <t.lukawski@samsung.com>
+ */
+(function (document, ns) {
+	"use strict";
+	//>>excludeStart("tauBuildExclude", pragmas.tauBuildExclude);
+	define(
+		[
+			"../../engine",
+			"../../event",
+			"../../util/selectors",
+			"../BaseWidget",
+			"../core",
+			"./Marquee"
+		],
+		function () {
+			//>>excludeEnd("tauBuildExclude");
+			var BaseWidget = ns.widget.BaseWidget,
+				engine = ns.engine,
+				utilSelectors = ns.util.selectors,
+				FILE_REGEXP = /[^\/]+$/,
+				classes = {
+					/**
+					 * Standard widget
+					 * @style ui-main-tab
+					 * @member ns.widget.core.MainTab
+					 */
+					MAIN_TAB: "ui-main-tab",
+					/**
+					 * Set Main Tab widget as active (visible)
+					 * @style ui-main-tab-active
+					 * @member ns.widget.core.MainTab
+					 */
+					ACTIVE: "ui-active",
+					/**
+					 * Set Tab as active
+					 * @style ui-tab-active
+					 * @member ns.widget.core.MainTab
+					 */
+					ACTIVE_TAB: "ui-tab-active"
+				},
+				MainTab = function () {
+					this._ui = {};
+				},
+				prototype = new BaseWidget();
+
+			MainTab.classes = classes;
+			MainTab.prototype = prototype;
+
+			/**
+			 * Build MainTab component
+			 * @method _build
+			 * @param {HTMLElement} element
+			 * @return {HTMLElement} Returns built element
+			 * @member ns.widget.core.MainTab
+			 * @protected
+			 */
+			prototype._build = function (element) {
+				element.classList.add(classes.MAIN_TAB);
+
+				return element;
+			};
+
+			/**
+			 * Init MainTab component
+			 * @method _init
+			 * @param {HTMLElement} element
+			 * @return {HTMLElement} Returns built element
+			 * @member ns.widget.core.MainTab
+			 * @protected
+			 */
+			prototype._init = function (element) {
+				var self = this,
+					pageContainer = null;
+
+				self._ui.parentPage = utilSelectors.getClosestBySelector(element, ".ui-page");
+
+				self._ui.links = [].slice.call(element.querySelectorAll("li > a"));
+				self._ui.links.forEach(function (link) {
+					var wrap = document.createElement("span"),
+						content = link.textContent.trim();
+
+					wrap.textContent = content;
+					link.textContent = "";
+					link.appendChild(wrap);
+
+					ns.widget.Marquee(wrap, {
+						iteration: 1,
+						delay: 0,
+						marqueeStyle: "endToEnd",
+						autoRun: false
+					});
+					link.setAttribute("data-rel", "maintab");
+				});
+
+				element.classList.add(classes.ACTIVE);
+
+				// change parent to pageContainer instead of page
+				pageContainer = utilSelectors.getClosestBySelector(element, ".ui-page-container");
+				pageContainer.appendChild(element);
+
+				return element;
+			};
+
+			/**
+			 * Handle events
+			 * @param {Event} event
+			 * @method handleEvent
+			 * @protected
```

Comment:
handleEvent is @public method.

Reply:
This is intended. This method is protected in widget context even that its name doesn't fit to name convention. The 'handleEvent' is special method for object which listens on events (https://developer.mozilla.org/en-US/docs/Web/API/EventListener/handleEvent). I cannot prefixed this method like "_handleEvent" and I cannot treated this method as public.

Second thing, we don't mark public method by `@public`.  We use only `@protected` or `@private` in jsDoc

Reply:
Okay. If you are right, we need to change the all @public to the @protected or @private.
I marked the handleEvent as @public before because the handleEvent was marked as @public.